astropy/modeling/parameters.py

```diff
@@ -54,6 +67,56 @@ def _tofloat(value):
     return value
 
 
+# Helpers for implementing operator overloading on Parameter
+def _binary_arithmetic_operation(op, right=False):
+    @functools.wraps(op)
+    def wrapper(self, val):
+        if self._model is None:
+            return NotImplemented
+
+        if self.unit is not None:
+            self_value = Quantity(self.value, self.unit)
+        else:
+            self_value = self.value
+
+        if right:
+            return op(val, self_value)
+        else:
+            return op(self_value, val)
+
+    return wrapper
+
+
+def _binary_comparison_operation(op):
+    @functools.wraps(op)
+    def wrapper(self, val):
+        if self._model is None:
+            return NotImplemented
+
+        if self.unit is not None:
+            self_value = Quantity(self.value, self.unit)
+        else:
+            self_value = self.value
+
+        return np.all(op(self_value, val))
```

Comment:
I think I'll change this in a separate PR, but I'll update it here as well once I've done that.